src/ApparentHorizons/HorizonManagerComponent.hpp

```diff
@@ -0,0 +1,76 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include "ApparentHorizons/HorizonManagerComponentActions.hpp"
+#include "DataStructures/DataBox/DataBox.hpp"
+#include "Parallel/ConstGlobalCache.hpp"
+#include "Parallel/Invoke.hpp"
+
+namespace HorizonManager {
+/// Holds all the options for a HorizonManager, so that
+/// they can be specified under a HorizonManager heading in the input file.
+struct OptionHolder {
+  struct Verbosity {
+    using type = size_t;
+    static constexpr OptionString help = {"Verbosity"};
+  };
+  using options = tmpl::list<Verbosity>;
+  static constexpr OptionString help = {"Options for horizon manager"};
+
+  explicit OptionHolder(size_t&& verbosity_in) : verbosity(verbosity_in) {}
+
+  OptionHolder() = default;
+  OptionHolder(const OptionHolder& /*rhs*/) = default;
+  OptionHolder& operator=(const OptionHolder& /*rhs*/) = default;
+  OptionHolder(OptionHolder&& /*rhs*/) noexcept = default;
+  OptionHolder& operator=(OptionHolder&& /*rhs*/) noexcept = default;
+  ~OptionHolder() = default;
+
+  size_t verbosity{};
+};
+}  // namespace HorizonManager
+
+namespace OptionTags {
+struct HorizonManager {
+  using type = ::HorizonManager::OptionHolder;
+  static constexpr OptionString help = {"Options for horizon manager"};
+};
+}  // namespace OptionTags
+
+/// Group component responsible for interpolating data HorizonComponents.
+/// Metavariables must contain (in addition to the usual things it contains)
+/// a type alias called horizon_tags that is a tmpl::list of AhTags,
+/// where each AhTag is a struct that contains:
+///         - a static function 'label()' returning std::string or const char*.
+///         - a type alias 'frame' to the ::Frame of the horizon.
+///         - a type alias 'option_tag' to something in Horizon::OptionTags.
+/// There should be one AhTag for each HorizonComponent.
+template <class Metavariables>
+struct HorizonManagerComponent {
+  using chare_type = Parallel::Algorithms::Group;
```

Comment:
I should document this better. The idea is that each Element sends its volume data only to the HorizonManagerComponent that lives on the same core as the Element, so that it's local communication.  Then after sending their data, the Elements go back to doing important things like computing right-hand sides for time steps, which is done concurrently with whatever horizon stuff is happening.  

The HorizonManagerComponents are responsible for interpolating data and communicating with the HorizonComponents.  Because horizon finding is iterative, there will typically be a handful of iterations for each horizon search, and each iteration requires communication back and forth between HorizonComponents and HorizonManagerComponents. This is done without any involvement of the Elements (except for the initial communication that needs to be done once for every horizon search, but not once for every iteration for every horizon search).

Reply:
Makes sense that elements just send their data and then keep computing other stuff. But could you use a singleton for collecting data and/or an array chare for parallelizing the horizon finding instead of the group chare? Or is it necessary to work on a particular core here? My current understanding is that we should leave the load balancing mostly up to Charm++.

Reply:
let's keep it as a group for now until we actually run something; a singleton would be overwhelmed by too much data.